lib/api/pokedex_api_spec/api_client.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:dio/dio.dart';
+import 'package:pokedex/utils/constant.dart' as k;
+
+class ApiClient {
+  late Dio dio;
+  static const String baseUrl = k.pokemonUrl;
```

Comment:
this is already a constant right? why make another constant to assign value for this constant? 🤔 

would it be better to already pass the value here immediately? ex => `static const String baseUrl = 'https://pokeapi.co/api/v2';` (no need to include the / at the of v2 so for later usage this will look like this => `'${baseUri.path}/pokemon/?limit=$_limitAmount'`)?